lib/cloudregister/registerutils.py

```diff
@@ -1635,3 +1644,25 @@ def __replace_url_target(config_files, new_smt):
                     current_service_server,
                     new_smt.get_FQDN())
                 )
+
+
+# -----------------------------------------------------------------------------
+def __connection_check(smt_ip_addr, ip_format):
+    if not smt_ip_addr:
+        # in case RMT ever dropped IPv4 support
+        return False
+    logging.info('Attempt to access update server over {}'.format(ip_format))
+    protocol = 'http'  # Default for backward compatibility
+    if https_only(get_config()):
+        protocol = 'https'
+    try:
+        cert_res = requests.get(
+            '%s://%s/smt.crt' % (protocol, smt_ip_addr),
```

Comment:
This breaks if the IP address has IPv6 format in that case `[]` are required.

Reply:
Well, in this PR, the call to this method adds the [] but I agree the method should add it (its responsibility) to remove the possibility of the caller forgetting to do it